test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -8128,18 +8128,18 @@ def expect(self, obj_type, attrib=None, id=None, op=EQ, attrop=PTL_AND,
 
         # Add check for substate=42 for jobstate=R, if not added explicitly.
         if obj_type == JOB:
-            add_attribs = {'substate': False}
+            add_attribs = {}
             substate = False
             for k, v in attrib.items():
                 if k == 'job_state' and ((isinstance(v, tuple) and
                                           'R' in v[-1]) or v == 'R'):
                     add_attribs['substate'] = 42
                 elif k == 'job_state=R':
-                    add_attribs['substate=42'] = v
+                    attrib['substate=42'] = v
```

Comment:
you should be setting add_attrib here instead of setting attrib because you are already updating attrib on line 8142